packages/slice-machine/components/CustomTypeTable/ctPage.tsx

```diff
@@ -1,98 +1,120 @@
 import { StatusBadge } from "../StatusBadge";
 import Link from "next/link";
-import React from "react";
+import React, { useState } from "react";
 import { Box, Text } from "theme-ui";
 import { FrontEndCustomType } from "@src/modules/availableCustomTypes/types";
 import { useModelStatus } from "@src/hooks/useModelStatus";
 import { KebabMenuDropdown } from "@components/KebabMenuDropdown";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import { RenameCustomTypeModal } from "@components/Forms/RenameCustomTypeModal";
 
 export const CustomTypeTable: React.FC<{
   customTypes: FrontEndCustomType[];
 }> = ({ customTypes }) => {
   const { modelsStatuses, authStatus, isOnline } = useModelStatus(customTypes);
+  const [renameCustomTypeName, setRenameCustomTypeName] = useState<string>("");
+  const [renameCustomTypeId, setRenameCustomTypeId] = useState<string>("");
```

Comment:
I'm not super happy with this way of doing things. But other than adding more complexity to the global state (in order to pass these through) I'm not sure how else to do it 😕